### ISSUE TITLE
Activity - DAI txns appearing under ETH activity

### ISSUE THREAD

**Issue by mr-michael** [tallycash/extension#588]
### What browsers are you seeing the problem on?

Chrome

### What were you trying to do?

View ETH txn history

### What did not work?

A large # of DAI txns appeared mixed in (curiously - only DAI but not txns from other tokens:

<img width="376" alt="Screen Shot 2021-12-11 at 5 52 25 PM" src="https://user-images.githubusercontent.com/7005061/145694302-48601390-4b48-407f-9565-fc19233bdb56.png">

Here's one txn that appeared in ETH activity history https://etherscan.io/tx/0xba3c8c9e51fc121f5e5c9f0861f2c773c6992da1f99a3d1e4290ecc41a581fe4

can add this read-only acct to confirm:
0x00De4B13153673BCAE2616b67bf822500d325Fc3

**... I'm wondering if the ETH-specific history is simply displaying global activity (on a spot-check, it looks like that's what may be happening)**

**Also worth noting that no other activity except ETH loads, including DAI**

<img width="355" alt="Screen Shot 2021-12-11 at 6 25 29 PM" src="https://user-images.githubusercontent.com/7005061/145694700-5726f316-388e-49cc-a2d7-3252980085f2.png">

**Comment by Shadowfiend** [comment]
Yep, here's the offending bit:

https://github.com/tallycash/extension/blob/875cdf583d25ae70be98e54c669e51796f3fcd5f/ui/pages/SingleAsset.tsx#L28-L35

`activity.asset` will always be ETH for Ethereum transactions; this needs to be switched to something like:

```typescript
  const filteredActivities = useBackgroundSelector((state) =>
    (selectCurrentAccountActivitiesWithTimestamps(state) ?? []).filter(
      (activity) => {
        if (typeof location.state.contractAddress !== "undefined") {
          location.state.contractAddress === activity.to
        } else if (activity.contractInfo?.type === "asset-transfer") {
          return activity.contractInfo.assetAmount.asset.symbol === symbol
        } else if (activity.contractInfo?.type === "asset-swap") {
          return activity.contractInfo.fromAssetAmount.asset.symbol === symbol ||
                 activity.contractInfo.toAssetAmount.asset.symbol === symbol
        }
        return false
      }
    )
  )
```

There's also other cleanup needed here, like probably `SingleAsset` should maybe be passed an `Asset`  in the first place, probably we should leverage asset similarity as used in the assets slice, and probably we need to externalize the “is this activity associated with this asset” function call as a separate utility. None of those are immediate requirements, however.

**Comment by mr-michael** [comment]
I'm continuing to not see any token history (only ETH):

https://etherscan.io/tx/0x3d26777152d73f42871298e0f6b4339bbb721a6e0c13c716113026304b877162

<img width="369" alt="Screen Shot 2021-12-15 at 11 22 49 AM" src="https://user-images.githubusercontent.com/7005061/146224531-299f54a7-a048-4f9b-8d84-b08d31f8b0e3.png">

<img width="384" alt="Screen Shot 2021-12-15 at 11 22 53 AM" src="https://user-images.githubusercontent.com/7005061/146224532-23aaf947-fcb4-4295-8b2c-c4554cbcabdc.png">

<img width="394" alt="Screen Shot 2021-12-15 at 11 22 41 AM" src="https://user-images.githubusercontent.com/7005061/146224534-bde9f29b-cc3c-4a06-9df0-e1abe6069f6a.png">

<img width="399" alt="Screen Shot 2021-12-15 at 11 22 45 AM" src="https://user-images.githubusercontent.com/7005061/146224536-b52861a9-fbd4-4f9d-bd1f-cdcba430a4de.png">

**Comment by henryboldi** [comment]
(Not closed by #670)
